src/service/gif.js

```diff
@@ -1,9 +1,13 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(
+        search?.trim() !== ''
+          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+          : '/chat/gif/featured'
+      )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `getGifFeatured` function is good as it now supports search functionality. However, there's no error handling in case the API request fails. It would be better to add a `.catch` block to handle any potential errors that might occur during the API call.
```diff
  export const getGifFeatured = async (search) =>
     new Promise((resolve, reject) => {
       api
        .get(
          search?.trim() !== ''
            ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
            : '/chat/gif/featured'
        )
        .then((res) => {
          resolve(res.data);
        })
+       .catch((error) => {
+         reject(error);
+       });
     });
```
This way, if the API call fails for any reason, the error will be caught and passed to the caller, allowing for more robust error handling.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
No need